### ISSUE TITLE
Don't guess when to format a partition

### ISSUE THREAD

**Issue by a1exsh** [zalando-stups/taupage#352]
After some discussion about recent issues related to unreliable formatting of partitions on startup we've decided that we don't want to rely on guessing for that.

This is only relevant for EBS volumes with flag `erase_on_boot` set to `False` (and instance storage needs to be formatted on every start anyway).  Given that we can require that EBS volumes to be formatted by Taupage should be tagged specifically.  After formatting, Taupage will remove the tag, so the volume can no longer be formatted by accident.

There remains a possibility that a volume that was created with the 'to be formatted' tag is picked up by an earlier version of Taupage, which doesn't honor the tag.  In order for this not to cause formatting when updating the AMI to the version which supports it, we might want to put the AMI version in the tag value, like so:
```
          TagName             |      TagValue      
------------------------------+--------------------
Taupage:format-if-ami-version | 2016-12-22 09:42:31
```
So that the volume will be formatted only if Taupage version matches exactly.  Older versions don't know about the tag, newer versions will just remove the tag without actually formatting.

**Comment by szuecs** [comment]
I like the idea, but I would like to have not the "Taupage version matches exactly", but a more future proof version, for example: "Taupage is newer than x".

Maybe you can get the "Create date/timestamp" for the image and do a >= known-version-that-works-as-expected?

**Comment by a1exsh** [comment]
The problem with "Taupage is newer than X" is exactly this: when you update the AMI, the version is newer than anything you knew about then creating the volume.  So if the format tag is still present, the "newer than" condition is always satisfied.

I imagine we will change our planb-cassandra deployment script so that it creates the EBS volumes and tags them explicitly like proposed here.  If someone deploys a cluster while new Taupage version implementing the feature is not yet released, the risk for them is that their volumes will be formatted when they update AMI to a version that implements it.

**Comment by szuecs** [comment]
Then I misunderstood your proposal. 
You want to TAG the volume, right?
What is the meaning of the TagValue for you?

**Comment by a1exsh** [comment]
@szuecs what I mean is that simply having a tag with any value like `True` will not do the trick.  Because it is possible that someone will deploy an instance with a version of Taupage which doesn't respect the tag yet, so it will not be removed.  Now, when the AMI is updated to a version that supports the tag, if it just looks for presence of the tag it will wrongly format the volume at this point.

So the way to avoid that which I propose is to include something in the tag value that will let tell apart these cases and don't do the wrong thing in the latter case.  I think the AMI version/timestamp is a natural choice.

**Comment by szuecs** [comment]
@a1exsh I totally agree!

**Comment by Raffo** [comment]
@a1exsh I also agree! Looking forward for the PR :-)

**Comment by a1exsh** [comment]
This should be fixed with #356

**Comment by Raffo** [comment]
This is fixed and documented in https://github.com/zalando-stups/documentation/pull/111
